ValidationRelay/Relay.swift

```diff
@@ -78,34 +92,36 @@ class RelayConnectionManager: ObservableObject {
     func triggerReconnect() {
         logItems.log("Triggering reconnect")
         connectionStatusMessage = "Reconnecting..."
-        // Delete the delegate so that more errors don't come in
         connectionDelegate?.disconnect()
         connectionDelegate = nil
         print("Waiting for \(backoff) backoff seconds")
         
         reconnectWork?.cancel()
         reconnectWork = nil
         
-        let work = DispatchWorkItem {
+        let work = DispatchWorkItem { [weak self] in
+            guard let self = self else { return }
             self.connectionDelegate = RelayConnectionDelegate(manager: self)
         }
-        // Wait a bit then try again
-        DispatchQueue.main.asyncAfter(deadline: .now() + DispatchTimeInterval.seconds(backoff), execute: work)
+        
+        DispatchQueue.main.asyncAfter(deadline: .now() + .seconds(backoff), execute: work)
         
         reconnectWork = work
-        backoff *= 2
+        backoff = min(backoff * 2, maxBackoff)
     }
 }
 
 class RelayConnectionDelegate: WebSocketConnectionDelegate, ObservableObject {
     var connection: WebSocketConnection
-    let manager: RelayConnectionManager
+    weak var manager: RelayConnectionManager?
 
-    init(
-        manager: RelayConnectionManager
-    ) {
+    init(manager: RelayConnectionManager) {
         self.manager = manager
-        connection = NWWebSocket(url: manager.currentURL!, connectAutomatically: true)
+        guard let currentURL = manager.currentURL else {
+            manager?.logItems.log("Current URL is nil", isError: true)
```

Comment:
```swift
Relay.swift:121:20: error: cannot use optional chaining on non-optional value of type 'RelayConnectionManager'
            manager?.logItems.log("Current URL is nil", isError: true)
            ~~~~~~~^
```
`manager` is a local variable here (from the init argument) that is non-optional